### DISCUSSION TITLE
How to change Image Extend/cropping

### DISCUSSION THREAD

**Discussion by MustafaElshani** [giotto-suite/Giotto#1067] [Q&A] [answered]
Currently when subsetting or after filtering spatPlot shows the image up until nearest spot. How can I change this so that it shows the image as per spatial coordinates? I can see here that you have an option for prefer but how and where do we determine this its seems polygon is the defaults https://github.com/drieslab/GiottoVisuals/blob/9931ad26aacbafcbcd2c5e1e30a5c0417fb32a8a/R/gg_info_layers.R#L1400

**Comment by jiajic** [comment, marked as answer]
You can currently only define the coordinate xy limits of the plot when using `spatInSituPlotPoints()`

Example:
```r
g <- GiottoData::loadGiottoMini("vis")
g <- subsetGiottoLocs(g, x_min = 4000, x_max = 5000, y_min = -4000, y_max = -3000)
spatInSituPlotPoints(g, show_image = TRUE, image_name = "image")
```
![image](https://github.com/user-attachments/assets/1f033a37-ccad-425d-86f8-46f38b1697cf)

```r
spatInSituPlotPoints(g, show_image = T, image_name = "image", xlim = c(3000, 6000), ylim = c(-5000, -2000))
```
![image](https://github.com/user-attachments/assets/96d3926a-4856-4f24-8348-c816ee06bbba)

We are trying to bring easier FOV setting to more of the plotting functions.

The `prefer` arg you point out is not quite what you need. It is passed to `ext()` and is currently only used with `giotto` objects in order to better select the desired spatial extent when multiple types of spatial information are present. Either the combined spatial bounds of all spatial information within the object (default) or specific subset types of spatial information. It is hardcoded to determine extent first from polygonal information, then spatial location (centroids, not user defined values), then feature points.

**Comment by MustafaElshani** [comment]
Hi George 

Great that works perfectly now I use `spatInSituPlotPoints` and I can crop/subset the exact FOV without loosing any of the image area, great for edges or TMA or small tumoroids which is my case.

now I use 
```
spatInSituPlotPoints(
  SpatialTrans_DMSO_ONTgene, 
  background_color = "white",
  polygon_fill = "leiden_clus",
  polygon_fill_as_factor = TRUE,
  show_image = TRUE,
  save_plot = FALSE,
  show_legend = FALSE,
  xlim = c(13800, 18300), 
  ylim = c(-17000, -14000)
```

Now if I want to show same area but with gene expression such as we do in `spatFeatPlot2D()` how would we approach that ?

**Reply by jiajic** [reply]
Right, this is something we hope to make easier with the other visualization functions in the near future. For now, the workaround is to append the information to the cell metadata.

Here is an example workaround function

```r
g <- GiottoData::loadSubObjectMini("visium")

spatinsitu_feat_workaround <- function(
        gobject, feat, 
        expression_values = "normalized", 
        spat_unit = NULL, 
        feat_type = NULL, 
        ...) {
    expr_table <- spatValues(gobject,
                             spat_unit = spat_unit, 
                             feat_type = feat_type,
                             expression_values = expression_values,
                             feats = feat)
    gobject <- addCellMetadata(gobject,
                               spat_unit = spat_unit,
                               feat_type = feat_type,
                               new_metadata = expr_table,
                               by_column = TRUE)
    p <- list(
        gobject = gobject,
        polygon_fill = feat,
        ...
    )
    do.call(spatInSituPlotPoints, p)
}
```
```r
spatinsitu_feat_workaround(g, 
    feat = "Gfap",
    show_image = T, 
    image_name = "image", 
    xlim = c(3000, 6000), 
    ylim = c(-5000, -2000)
)
```
![image](https://github.com/user-attachments/assets/e04de879-fc56-446f-a8e6-fa68f76aa031)

**Reply by MustafaElshani** [reply]
Thank you this works great and was exactly what I have been looking for. Appreciated
